tools/mggit/tools.go

```diff
@@ -25,21 +25,22 @@ func VerifyNoDiff(ctx context.Context) error {
 	return nil
 }
 
-func Version(ctx context.Context) string {
-	cmd := mgtool.Command(ctx, "git", "rev-parse", "--verify", "HEAD")
-	var b bytes.Buffer
-	cmd.Stdout = &b
-	if err := cmd.Run(); err != nil {
-		panic(err)
+// RunCommand runs git with the given arguments, and returning the output as a string.
+// It will panic if an error occurs.
+func RunCommand(ctx context.Context, args ...string) string {
```

Comment:
Feels like this one deviating a bit from all the other “Command” apis we have, where they generally returns a exec.Cmd . However, it feels like this package is more of a helper one, then a tool. Should it be moved some place else? Given we are breaking the pattern with it. For it to remain here it should really just be a Command that returns git as a exec.Cmd

Reply:
To add, not saying what you added here is bad 😅 I just feel we should write down the idea we have with the general design, so it’s easier to reason about stuff like this :) fyi @odsod 
Let’s add some design philosophies for the repo?

Reply:
Yes, I thought about that too and felt this one warranted a side-step from the convention. With that said it should probably be renamed to "Run" if we decide to keep it.

The reason I added this function without the `exec.Command` is that you can do this:
```go
	var (
		gitTag    = mggit.RunCommand(ctx, "describe", "--tags")
		gitCommit = mggit.RunCommand(ctx, "log", "-1", "--pretty=format:%H")
		gitAuthor = mggit.RunCommand(ctx, "show", "-s", "--format=%an <%ae>", gitCommit)
		gitBranch = mggit.RunCommand(ctx, "rev-parse", "--abbrev-ref", "HEAD")
	)

```
which would be major hassle with the command pattern used elsewhere.

Reply:
This looks more in line with the `GetOutput` function of the exec package - maybe we can find a name that points more in that direction?

Reply:
I think we should rather add a helper function that takes an exec.Command and returns the stdout output and panics if there's an error. That can be combined with the conventional Command functions from all the packages?

Reply:
Yes, I was thinking you could have two conventions, one of returning an exec.Command and one that is a convenience layer on top of that. 

Maybe straight up a `MustOutput` ? or simply `Output`?

Reply:
hm good point, then it could be re-used elsewhere. I'll see if I can try something out